### DISCUSSION TITLE
Option to accept integers as floats?

### DISCUSSION THREAD

**Discussion by felixchenier** [beartype/beartype#173] [Q&A]
Good morning

I'm trying to see if beartype would fit my need as a simple-to-operate real-time type checker. This is for a public API that I'm building, and I don't want to manually check everything, or to reprogram something that is for sure a very deep rabbit hole!

But I have an issue with beartype being a little bit too rigorous. Many of my functions accept floats as parameters, and I expect the users (including me) to sometimes provide integers instead of floats. The calculations work, and mypy is also fine with it. However, by using beartype, I have a runtime error (int != float), which in my case is not what I would need.

I don't want to mess with every signature and change (a: float) to (a: Union[float, int]), which:
1. is ugly and useless;
2. will fail eventually, since I will forget one, mypy won't tell me, but I'll have plenty of users who will have unexpected runtime errors.

Is there an easy way to switch off some type checks, such as this one?

Thank you!

**Comment by TeamSpen210** [comment]
According to [PEP 484](https://peps.python.org/pep-0484/#the-numeric-tower) this is actually a special case, int should be treated as acceptable to pass to a float parameter. The types aren't actually 100% compatible, but practically speaking it's good enough and this is common to do. That's why MyPy doesn't complain.

**Reply by felixchenier** [reply]
Thanks for the answer, it confirms that I'm right to expect ints to work when floats are expected. My question was more related to beartype rather than mypy. Is there a configuration option to tell beartype to accept this special case, because at the moment it generates an error when an int is provided instead of a float.

**Comment by leycec** [comment]
**Excellent feature request,** @felixchenier. So excellent, in fact, that I've reposessed your initial question into a [new feature request at our issue tracker](https://github.com/beartype/beartype/issues/174). It's now official, folks. We're doin' this. Eventually.

## tl;dr

@beartype should support this... but currently doesn't:

```python
from beartype import beartype, BeartypeConf

beartype_towering = beartype(conf=BeartypeConf(is_coerce_numbers=True))
'''
:func:`beartype.beartype` decorator configured to enable the
:pep:`484`-compliant **implicit numeric tower** (i.e., allow lossy coercions
from `complex` to `float` to `int` types).
'''

@beartype_towering
def crunch_numbers(numbers: list[float]) -> float:
    return sum(numbers)

# This is now fine.
crunch_numbers([3, 1, 4, 1, 5, 9])

# This is still fine, too.
crunch_numbers([3.1, 4.1, 5.9])
```

Let's continue the discussion at #174, if it wouldn't be too much trouble. And thanks again, @felixchenier. I love fielding fascinating typing questions like this. :smile:
